### PR TITLE
added colorization to negative bar chart

### LINKED ISSUE #322: Improvements for FeatureImportances visualizer
There are a couple of enhancements for the `yellowbrick.features.FeatureImportances` visualizer that should be made to really make it stick out. They are as follows:

*Note to contributors: items in the below checklist don't need to be completed in a single PR; if you see one that catches your eye, feel to pick it off the list!*

- [ ] color negative coefs 
- [ ] top n features to filter number displayed (both pos and neg)
- [ ] implement standard deviation for ensemble models 

### Color negative coefs 

The first item is relatively straightforward, currently, the bar chart is a single color, but it might be nice to show negative `coef_` values as a different color, e.g. blue for positive green for negative as below:

figure_1

To do this, you'll have to create a color array to pass as the color argument, e.g. 

```python
colors = np.array(['b' if v > 0 else 'g' for v in self.feature_importances_]) 
self.ax.barh(pos, self.feature_importances_, color=colors, align='center')
```
We should also create arguments to provide a way to specify the colors. 

### Top N Features 

For the second item, I'm picturing something similar to [most informative features with scikit-learn](https://stackoverflow.com/questions/11116697/how-to-get-most-informative-features-for-scikit-learn-classifiers) (though not exactly this code). Here, an argument `topn` which defaults to None specifies a filter to only plot the `N` best features. 

This should also be relatively straightforward, but gets complicated in the case of negative values. We have two options, we can rank all values including negative values and plot the N best values either positive or negative, or we can do the N best positive and N best negative coefficients 

### Standard Deviation for Ensembles 

Ensemble models like Random Forest and Gradient Boosting have an underlying `estimators_` attribute that describes each feature's importance in a different way. The global feature importances are the mean, but it would be nice to add an `xerr` bar with the standard deviation as in [plot forest importances](http://scikit-learn.org/stable/auto_examples/ensemble/plot_forest_importances.html). 

This could also be useful for `CV` models that also have an underlying `estimators_` attribute. 

The idea with this one is to compute the standard deviations for each feature using `estimators_` and np.std, and storing the value in a `confidences_` attribute during fit. Note that it will also have to be sorted using the `sort_index` -- the confidences are drawn during `ax.barh` with `xerr=self.confidences_`. 

Right now it looks like the example above is no longer working exactly as expected, so some deeper review is necessary. 

See also #194 where a discussion about tree-specific feature importances is ongoing.

### PR DESCRIPTION
**Pull request by kim81682**
Fixes #322 

[Description]
addressed just the first item
Changes proposed in this pull request:

-the previous bar graph was commented out. 2 additional lines were added: one for color, the second for graphing the color if the item is negative.
-
-

@DistrictDataLabs/team-oz-maintainers

### CODE REVIEW

File: yellowbrick/features/importances.py [hunk 1]
```diff
@@ -168,7 +171,12 @@ def draw(self, **kwargs):
         pos = np.arange(self.features_.shape[0]) + 0.5
 
         # Plot the bar chart
-        self.ax.barh(pos, self.feature_importances_, align='center')
+        #raise Exception(self.feature_importances_)
+        # Plot the bar chart
+
+
+        colors = np.array([self.color_pos if v > 0 else self.color_neg for v in self.feature_importances_])
```
Comment (kim81682) [inline yellowbrick/features/importances.py]: Fixes "color negative coefs" item

**Comment by lwgray** [comment]
Thank you for joining us for the Yellowbrick Sprint at PyCon Cleveland. And also, thank you for submitting your first PullRequest - oh yeah! 

We will review this asap. We have been quite busy with all the great contributions from sprinters like yourself.  In the meantime, feel free to work on another issue.  We hope you continue to contribute.

File: yellowbrick/features/importances.py [hunk 2]
```diff
@@ -168,7 +171,12 @@ def draw(self, **kwargs):
         pos = np.arange(self.features_.shape[0]) + 0.5
 
         # Plot the bar chart
-        self.ax.barh(pos, self.feature_importances_, align='center')
+        #raise Exception(self.feature_importances_)
```
Comment (ndanielsen) [inline yellowbrick/features/importances.py]: Thank you so much for working with on during sprints! I really enjoyed meeting you! One item to fix here, can you remove these comments?

File: yellowbrick/features/importances.py [hunk 3]
```diff
@@ -86,7 +86,7 @@ class FeatureImportances(ModelVisualizer):
     >>> visualizer.poof()
     """
 
-    def __init__(self, model, ax=None, labels=None, relative=True,
+    def __init__(self, model, ax=None, labels=None, relative=True, color_neg='b', color_pos='g',
```
Comment (ndanielsen) [inline yellowbrick/features/importances.py]: One minor change, can you please update the documentation for this visualizer class? It should be a pretty straightforward change located in this section of the code:

https://github.com/kim81682/yellowbrick/blob/3489dd1a7b97901919e1bc2fcbafbaa8a5a20cc9/yellowbrick/features/importances.py#L55-L70

Again, thanks for contributing this!

**Comment by kim81682** [comment]
Absolutely! Thank you for the experience. I still have a great deal to learn but would love to continue contributing/learning with your team :)
~kin


Sent with BlackBerry Work
(www.blackberry.com)

File: yellowbrick/features/importances.py (hunk 1 above)
Comment (lwgray) [inline yellowbrick/features/importances.py]: Could this line be changed to reflect that we consider Zero to be a positive number and should be colored as such.  

`if v >= 0`

**Comment by lwgray** [comment]
@kim81682  Is there anything we can do to help you complete this PR?

@ndanielsen

**Pull request closed without merging**